service/history/workflow/workflow_task_state_machine.go

```diff
@@ -769,3 +781,17 @@ func (m *workflowTaskStateMachine) emitWorkflowTaskAttemptStats(
 		)
 	}
 }
+
+func (m *workflowTaskStateMachine) getStartToCloseTimeout(
+	defaultTimeout time.Duration,
+	attempt int32,
+) time.Duration {
+	if attempt <= workflowTaskRetryBackoffMinAttempts {
+		return defaultTimeout
+	}
+
+	policy := backoff.NewExponentialRetryPolicy(workflowTaskRetryInitialInterval)
+	policy.SetMaximumInterval(m.ms.shard.GetConfig().WorkflowTaskRetryMaxInterval())
+	policy.SetExpirationInterval(backoff.NoInterval)
+	return defaultTimeout + policy.ComputeNextDelay(0, int(attempt)-workflowTaskRetryBackoffMinAttempts)
+}
```

Comment:
should the passive cluster receive workflow task timeout updates?

Reply:
Synced offline and added more comments in this function.